### PR TITLE
 add configuration options for table

### PR DESCRIPTION
**Pull request by albertlockett**
makes the index degree of the table configurable 

it can be set on the table config using the new method `WithIndexDegree(int)`
```golang
tableConfig := arcticdb.NewTableConfig(schema, 10, 512*1024*1024).WithIndexDegree(3)
```

### CODE REVIEW

**Review by brancz — COMMENTED** [review]
Really awesome, thank you for tackling this!

File: table.go [hunk 1]
```diff
@@ -455,7 +467,7 @@ func (t *TableBlock) splitGranule(granule *Granule) {
 		return
 	}
 
-	granules, err := g.split(tx, t.table.config.granuleSize/2) // TODO magic numbers
+	granules, err := g.split(tx, t.table.config.granuleSize/t.table.config.indexDegree)
```
[the remark this answers is not in the document]
Comment (albertlockett2) [reply table.go:470]: Sure @brancz sounds good. I wasn't sure if it was related either, I was kind of guessing. I'll change to split size

[the remark this answers is not in the document]
Comment (albertlockett) [reply table.go:470]: Sure @brancz sounds good. I wasn't sure if it was related either, I was kind of guessing. I'll change to split size

[the remark this answers is not in the document]
Comment (albertlockett) [reply table.go]: @brancz updated

**Comment by brancz** [comment]
Unfortunately, it seems we have introduced a merge conflict. Could you rebase and move the configurations like we discussed on discord? Otherwise, this looks awesome!

**Comment by metalmatze** [comment]
Seems like it's all green now. Want to merge @brancz ?

**Review by brancz — COMMENTED** [review]
Awesome!

**Pull request merged**